conformance/utils/suite/experimental_suite.go

```diff
@@ -83,51 +96,63 @@ type ExperimentalConformanceOptions struct {
 }
 
 // NewExperimentalConformanceTestSuite is a helper to use for creating a new ExperimentalConformanceTestSuite.
-func NewExperimentalConformanceTestSuite(s ExperimentalConformanceOptions) (*ExperimentalConformanceTestSuite, error) {
-	config.SetupTimeoutConfig(&s.TimeoutConfig)
+func NewExperimentalConformanceTestSuite(options ExperimentalConformanceOptions) (*ExperimentalConformanceTestSuite, error) {
+	config.SetupTimeoutConfig(&options.TimeoutConfig)
 
-	roundTripper := s.RoundTripper
+	roundTripper := options.RoundTripper
 	if roundTripper == nil {
-		roundTripper = &roundtripper.DefaultRoundTripper{Debug: s.Debug, TimeoutConfig: s.TimeoutConfig}
+		roundTripper = &roundtripper.DefaultRoundTripper{Debug: options.Debug, TimeoutConfig: options.TimeoutConfig}
+	}
+
+	installedCRDs := &apiextensionsv1.CustomResourceDefinitionList{}
+	err := options.Client.List(context.TODO(), installedCRDs)
+	if err != nil {
+		return nil, err
+	}
+	apiVersion, apiChannel, err := getAPIVersionAndChannel(installedCRDs.Items)
+	if err != nil {
+		return nil, err
```

Comment:
If I'm understanding this correctly, this means that it would be impossible to run conformance tests in a cluster with mismatched CRDs? I'm not sure we want to be so strict here, maybe we should present a warning and/or have some kind of output that means that CRDs from different bundle versions or release channels were used.

For example, imagine that someone wanted the GA version of HTTPRoute, Gateway, GatewayClass installed because they were using those in production, but also wanted to experiment with gRPCRoute for a different controller. It seems like this change would prevent anyone from running conformance tests in that cluster.

As an alternative, it may be viable to say that the only way you can print out a formal/verified conformance report is to have all CRDs match version and channel, but still allow anyone else to run tests with mismatched version/channel just to verify the results or do some form of spot testing.

Or one more option - this is really just metadata, if we can't find it, maybe we just don't populate it at all?

Reply:
I get your point @robscott, I've made the following change:
- default behavior: in case there is a mismatch between versions or channels, the suite fails.
- the option `AllowCRDsMismatch` allows to customize this behavior; in case it is set to false and there is a mismatch between versions and channels, the suite does not fail, and the value `UNDEFINED` is set in the `GatewayAPIVersion` and `GatewayAPIChannel` of the report. There is a corresponding flag to customize such options when using the CLI.
- I've added an additional check on the suite version. In case the CRDs version is different from the suite version, the suite fails (unless the `AllowCRDsMismatch` option is set).